openshift/jenkins/pims-api/Jenkinsfile

```diff
@@ -0,0 +1,161 @@
+// Implementation Details
+// The solr build depends on the solr-base build (more precisely the solr-base image stream)
+// which only needs to be rebuilt when there are changes to it's associated repository (https://github.com/bcgov/openshift-solr)
+// Therefore, it is unnecessary to build it as part of this pipeline.
+
+def PROJECT_NAMESPACE = 'jcxjin'
+def APP_NAME = 'pims'
+def COMP_NAME = 'api'
+def BUILD_ENV = 'tools'
+def CONTEXT_DIRECTORY = 'backend'
+def ENVIRONMENTS = ['dev', 'test']
+def TAG_NAMES = ['dev', 'test' ]
+def WAIT_TIMEOUT = 10 // minutes
+
+// Determine whether there were any changes the files within the project's context directory.
+@NonCPS
+boolean triggerBuild(String contextDirectory) {
```

Comment:
Would love to find a way to move shared library code to a single file somewhere and import it in.